contrib/udfs/src/main/java/org/apache/drill/exec/udfs/DateFunctions.java

```diff
@@ -140,8 +143,77 @@ public void eval() {
       java.time.format.DateTimeFormatter formatter = java.time.format.DateTimeFormatter.ofPattern(format);
       java.time.LocalDateTime dateTime = java.time.LocalDateTime.parse(inputDate, formatter);
 
-      java.time.LocalDateTime td = org.apache.drill.exec.udfs.NearestDateUtils.getDate(dateTime, intervalString);
+      java.time.LocalDateTime td = DateConversionUtils.getDate(dateTime, intervalString);
       out.value = td.atZone(java.time.ZoneId.of("UTC")).toInstant().toEpochMilli();
     }
   }
+
+  @FunctionTemplate(names = {"yearweek","year_week"},
+    scope = FunctionTemplate.FunctionScope.SIMPLE,
+    nulls = FunctionTemplate.NullHandling.NULL_IF_NULL)
+  public static class YearWeekFunction implements DrillSimpleFunc {
+    @Param
+    VarCharHolder inputHolder;
```

Comment:
Question here... My thinking in having the String input was that I wanted this function to be durable.  IE:  If a user provides strangely formatted dates, it still should work.  However, do you think it might make sense to remove the version which accepts a date as input?

Reply:
See below, I think that in that case we want this function defined for DATE only using the convertlet stuff to convert it to `100*year()+week()` and a new
```
create function INT year_week_auto_parse(date_str VARCHAR) as 100*year(to_timestamp(date_str, 'auto')) + week(to_timestamp(date_str, 'auto'));
```
(assuming we can do these pure SQL UDFs). So we're composing general purpose functions to get to the specific "What I need right now is a durable year_week function" instead of hardcoding different behaviour into a new `year_week` to what we offer for e.g. `year`, `month`, or `week`. The latter would leave users frustrated by date extraction functions that arbitrarily (to them) work on mixed format strings in some cases and not in others.

Reply:
@jnturton I removed the extra UDF.  I haven't messed with the convertlet logic before, but my initial digging seemed to imply that it was only for SQL keywords as defined by Calcite.  Do you know where in Drill I might add this?

Reply:
@vvysotskyi suggested DrillConvertletTable...